extensions/cornerstone/src/utils/mpr/toggleMPRHangingProtocol.ts

```diff
@@ -58,26 +58,19 @@ export default function toggleMPRHangingProtocol({
   const viewportDisplaySetInstanceUIDs =
     viewports[activeViewportIndex].displaySetInstanceUIDs;
 
-  const errorCallback = error => {
-    // Unable to create MPR, so be sure to return to the cached/original protocol.
-    hangingProtocolService.setProtocol(
-      cachedState.protocol.id,
-      viewportMatchDetails,
-      restoreErrorCallback
-    );
+  // What is the current active protocol and stage number to restore later
+  const { protocol, stage } = hangingProtocolService.getActiveProtocol();
 
+  const restoreErrorCallback = error => {
```

Comment:
Please dump the error message to console.log, as otherwise it can be hard to figure out why something went wrong, and it is too late to grab the error message.